eliot-service/eliot/libmarkus.py

```diff
@@ -40,3 +185,28 @@ def setup_metrics(app_config, logger=None):
         )
     markus.configure(markus_backends)
     _IS_MARKUS_SETUP = True
+
+
+class RegisteredMetricsFilter(MetricsFilter):
+    """Filter for enforcing registered metrics emission."""
+
+    def __init__(self, metrics):
+        self.metrics = metrics
+
+    def filter(self, record):
+        metric = self.metrics.get(record.key)
+        if metric is None:
+            LOGGER.warning("metrics key %r is unknown", record.key)
+
+        elif record.stat_type != metric.stat_type:
+            LOGGER.warning(
+                "metrics key %r has wrong type; got %s expecting %s",
+                record.key,
+                record.stat_type,
+                metric.stat_type,
```

Comment:
This doesn't validate the tags. I thought about that, but it seems too burdensome to do all the time.

Another thought I had was to turn the filter on and off depending on whether you're in a server environment so it's not doing extra work validating something that should get caught in development or testing.